### PR TITLE
[export] Update docs

### PR DESCRIPTION
**Pull request by angelayi**
Preview: https://docs-preview.pytorch.org/pytorch/pytorch/157750/export.html

Changes:
* Rename draft_export.md -> export.draft_export.md for consistency.
* Removed non-strict section in export, instead pointed to programming model doc.
* Extended "Expressing Dynamism" section to include Dim hints, ShapeCollection, and AdditionalInputs.
* Removed Specialization section in favor of programming model doc
* Added pt2 archive doc
* Cleaned up sidebar

### CODE REVIEW

File: docs/source/export.pt2_archive.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+│   └── ....json
+└── models
+    ├── model1.json
+    └── model2.json
+```
+
+## Contents
+
+### Archive Headers
+* `archive_format` declares the format used by this archive. Currently, it can only be “pt2”. This tag can be used to distinguish archives produced from TorchScript and PT2 Stack.
```
Comment (svekars) [inline docs/source/export.pt2_archive.md]: do we have to mention torchscript?

File: docs/source/export.draft_export.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-(draft-export)=
+(export.draft_export)=
 
 # Draft Export
```
Comment (svekars) [inline docs/source/export.draft_export.md]: do we want to rename the file? There are some ideas I have about restructuring all export/compiler docs that might eliminate the need to add prefix. You could also just create a dir called export and put all related docs there.

File: docs/source/export.md [hunk 3]
```diff
@@ -488,211 +320,239 @@ Some additional things to note:
...
+```python
+batch = Dim("batch")
+dynamic_shapes = {"x1": {0: dim}, "x2": {0: batch}}
 ```
 
+For each `Dim`, we can specify minimum and maximum values. We also allow
+specifying relations between `Dim`s in univariate linear expressions: `A * dim + B`.
+This allows users to specify more complex constraints like integer divisibility
+for dynamic dimensions. These features allow for users to place explicit
+restrictions on the dynamic behavior of the ExportedProgram produced.
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
restrictions on the dynamic behavior of the `ExportedProgram` produced.
```

File: docs/source/export.md [hunk 4]
```diff
@@ -488,211 +320,239 @@ Some additional things to note:
...
-{func}`torch.export.Dim` will need to be used to specify the dimension
-of the input tensor (`x.shape[0]`) to be dynamic, and the source code will
-need to be {ref}`rewritten <data-shape-dependent-control-flow>`.
+(training-export)=
+
+## Export IR, Decompositions
 
-Note that tensors that are part of the module state (e.g. parameters and
-buffers) always have static shapes.
+The graph produced by torch.export returns a graph containing only ATen
```
Comment (svekars) [inline docs/source/export.md]:
```suggestion
The graph produced by `torch.export` returns a graph containing only ATen
```

File: docs/source/export.draft_export.md (hunk 2 above)
Comment (angelayi) [reply docs/source/export.draft_export.md]: > You could also just create a dir called export and put all related docs there

Oh yeah I can do that!

**Review by pianpwk — APPROVED** [review]
